File: lib/mosek/11.0/tools/examples/dotnet/opt_server_sync.cs
/*
  Copyright: Copyright (c) MOSEK ApS, Denmark. All rights reserved.

  File:    opt_server_sync.cs

  Purpose :   Demonstrates how to use MOSEK OptServer
              to solve optimization problem synchronously
*/
using System;

namespace mosek.example
{
  class msgclass : mosek.Stream
  {
    public override void streamCB (string msg)
    {
      Console.Write ("{0}", msg);
    }
  }

  public class simple
  {
    public static void Main (string[] args)
    {
      if (args.Length == 0)
      {
        Console.WriteLine ("Missing arguments, syntax is:");
        Console.WriteLine ("  opt_server_sync inputfile addr [certpath]");
      }
      else
      {
        String inputfile = args[0];
        String addr      = args[1];
        String cert      = args.Length < 3 ? null : args[2];

        using (mosek.Task task = new mosek.Task())
        {
          task.set_Stream (mosek.streamtype.log, new msgclass ());

          // Load some data into the task
          task.readdata (inputfile);

          // Set OptServer URL
          task.putoptserverhost(addr);

          // Path to certificate, if any
          if (cert != null)
            task.putstrparam(sparam.remote_tls_cert_path, cert);

          // Optimize remotely, no access token
          mosek.rescode trm = task.optimize ();

          task.solutionsummary (mosek.streamtype.log);
        }
      }
    }
  }
}

File: lib/mosek/11.0/tools/examples/matlab/opt_server_sync.m
%
%  Copyright : Copyright (c) MOSEK ApS, Denmark. All rights reserved.
%
%  File :      opt_server_sync.m
%
%  Purpose : Demonstrates solving a problem remotely using the OptServer.
%
%            url should be 'http://server:port' or 'https://server:port'
%            cert is the path to a TLS certificate, if using https
%

function opt_server_sync(url, cert)

% Here we can set up a model
model = mosekmodel(...
              objsense = "min", ...
              objective = [ 1, 2 ]', ...
              numvar = 2, ...
              F = [ 1 0 ; ...
                    0 1 ; ...
                    3 -1 ], ...
              domain = [ mosekdomain("greater than", rhs = 2), ... 
                         mosekdomain("greater than", rhs = 3), ... 
                         mosekdomain("less than", rhs = 1) ]);     


% Set up the certificate path, if using TLS, otherwise ignore
if exist('cert','var')
	param = ["MSK_SPAR_REMOTE_TLS_CERT_PATH", cert ];
else
	param = [];
end

% Optimize using the remote server
model.solve(param = param, optserver = url);

% Use the optimal solution
if model.hassolution("interior")
	xx = model.getsolution("interior", "x");
	disp(xx);
end

